### DISCUSSION TITLE
Fix Docker Compose invalid interpolation format

### DISCUSSION THREAD

**Discussion by s-crypt** [jasonacox/Powerwall-Dashboard#458] [General]
If you are getting the error

`Invalid interpolation format for "influxdb" option in service "services": "${INFLUXDB_PORTS:-8086:8086}"`

when trying to run `docker compose`, make sure you uninstall `docker-compose` and verify `docker-compose-plugin` is up to date with apt or your package manager and the docker repository.

Restart your device and you should be able to run `docker compose` successfully.

I believe this has to do with `docker-compose` being deprecated in favor of docker compose v2, which uses `docker-compose-plugin`. 
Now that the version is removed from `powerwall.yml` in 4c2ae0f0c0541cab825546b8d8fe60a21469b4da, `docker-compose`(v1) cannot handle the file but the `docker-compose-plugin`(v2) can using `docker compose`

Hopefully this saves someone the hour that it took me to figure this out lol
Kudos to [this guy](https://github.com/opencv/cvat/issues/5936#issuecomment-1488867165) which appeared through my internet searching who proposed this solution on an entirely unrelated project.

**Comment by BJReplay** [comment]
Hey, really sorry it took you a while to figure that out.

Questions to try to help others avoid the problem you had: how did you come to remove the version from powerwall.yml without running the update script?

Or did you use the update script (update.sh)?

Reason I ask is that update.sh is supposed to check that you already have docker compose V2 installed, so that you shouldn't run into this problem.

There's a check in upgrade.sh to prompt you to upgrade before it updates to the latest version that removes the version tag from powerwall.yml.

**Reply by s-crypt** [reply]
I was running `docker compose` manually (more specifically `docker compose -f powerwall.yml up -d`), because after a successful update, I upgraded my grafana version in `powerwall.yml`.
The script works fine, as well as the upgrade. Running the command manually is where I experienced the issue of conflicting docker compose commands.
